### ISSUE TITLE
Provide more advanced code samples of more advanced scenarios

### ISSUE THREAD

**Issue by oskardudycz** [EventStore/documentation#166]
Based on @rmaziarka feedback - it would be good to add samples and how-to around:
- optimistic/pessimistic locking, 
- idempotent consumer, 

@rmaziarka feel free to put more comments or ideas 👍

**Comment by alexeyzimarev** [comment]
We discussed it before. However, we aren't sure where it belongs. Remember that this is documentation about our product. It's not an Event Sourcing guide or tutorial. It doesn't mean we don't want to create something like that, the question is if it belongs to the docs.

**Comment by oskardudycz** [comment]
From my perspective, people thinking about using the event store solution usually don't have good knowledge around Event Sourcing. From my experience, they're learning both the tool and the pattern at the same time. Plus people before choosing a solution are looking for an answer if and how it supports such concepts. I think that we should provide guidance around that and step by step introduce those concepts together with teaching the product.

In Marten, we have a concept of "scenarios" (https://martendb.io/documentation/scenarios/) - they have a separate section. However, I have mixed feeling about that, as this section is not easily discoverable.

I think that we should have it somewhere in the documentation (we can discuss the exact place), as if we move it to blog posts then it would be harder to find and people will need to jump between one place and another.

**Comment by alexeyzimarev** [comment]
I totally get it and we know that the issue exists and we want to address it. But it is not a _documentation_. We have to elaborate more on the educational direction, make some decisions and do it _after_ the documentation is in a decent shape for 20.6.1 (with clients).

**Comment by oskardudycz** [comment]
I agree. I’m not saying that we have to do it now. I added this ticket to have a track on that (especially that it came from the developer feedback).

I’m okay with keeping it open until we have a clear vision.
